### PR TITLE
soc: ambiq: Add power management support for Apollo3

### PR DESCRIPTION
**Pull request by AlessandroLuo**
Add power management support for Ambiq Apollo3 SoCs and drivers

### CODE REVIEW

File: drivers/serial/uart_pl011.c [hunk 1]
```diff
@@ -648,33 +658,30 @@ void pl011_isr(const struct device *dev)
...
-	};							\
-								\
-	DEVICE_DT_INST_DEFINE(n, pl011_init,			\
-			NULL,					\
-			&pl011_data_port_##n,			\
-			&pl011_cfg_port_##n,			\
-			PRE_KERNEL_1,				\
-			CONFIG_SERIAL_INIT_PRIORITY,		\
-			&pl011_driver_api);
+#define PL011_INIT(n)                                                                              \
```
Comment (RichardSWheatley) [inline drivers/serial/uart_pl011.c:661]: Why is this section being modified?

**Review by RichardSWheatley — DISMISSED** [review]

File: drivers/serial/uart_pl011.c (hunk 1 above)
Comment (AlessandroLuo) [reply drivers/serial/uart_pl011.c:661]: there was compliance check fail, this modification is due to clang-format

Comment (RichardSWheatley) [reply drivers/serial/uart_pl011.c:661]: I may have to do that with mine as well.

File: soc/ambiq/apollo3x/power.c [hunk 2]
```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (c) 2024 Ambiq LLC
```
Comment (aaronyegx) [inline soc/ambiq/apollo3x/power.c:2]: Should be Ambiq Micro?

**Review by aaronyegx — APPROVED** [review]
LGTM.

**Review by RichardSWheatley — APPROVED** [review]

**Pull request merged**